### PR TITLE
fix(order): fix the bug of panic when the order stop

### PR DESCRIPTION
**Pull request by Karenlrx**

### CODE REVIEW

File: pkg/peermgr/swarm.go [hunk 1]
```diff
@@ -136,35 +136,42 @@ func (swarm *Swarm) Start() error {
...
-		go func(id uint64, addr *peer.AddrInfo) {
+		go func(id uint64, addr *peer.AddrInfo, ctx context.Context) {
 			if err := retry.Retry(func(attempt uint) error {
-				// for restart node, after updating the routing table, some nodes may not exist in routing table
-				routers := swarm.notifiee.getPeers()
-				if _, ok := routers[id]; !ok {
-					swarm.logger.Infof("Can't find node %d from routing table, stopping connect", id)
+				select {
+				case <- swarm.ctx.Done():
+					swarm.cancel()
```
Comment (sandyzhou) [inline pkg/peermgr/swarm.go:143]: cancel() should have  been called before to trigger ctx.Done()

File: pkg/peermgr/swarm.go [hunk 2]
```diff
@@ -213,6 +220,7 @@ func (swarm *Swarm) Ping() {
 			ticker.Stop()
 			ticker = time.NewTicker(swarm.pingTimeout)
 		case <-swarm.ctx.Done():
+			swarm.cancel()
```
Comment (sandyzhou) [inline pkg/peermgr/swarm.go:223]: cancel() should have  been called before to trigger ctx.Done()

File: pkg/order/mempool/tx_cache.go [hunk 3]
```diff
@@ -111,3 +111,7 @@ func (tc *TxCache) stopTxSetTimer() {
 	close(tc.stopTimerC)
 	tc.stopTimerC = make(chan bool)
 }
+
+func (tc *TxCache) StopTxListen() {
+	tc.Close <- true
```
Comment (sandyzhou) [inline pkg/order/mempool/tx_cache.go:116]: close(tc.close) is enough

File: pkg/order/mempool/tx_cache.go [hunk 4]
```diff
@@ -20,7 +20,7 @@ type TxCache struct {
 	txSet      []pb.Transaction
 	timerC     chan bool
 	stopTimerC chan bool
-	close      chan bool
+	Close      chan bool
```
Comment (sandyzhou) [inline pkg/order/mempool/tx_cache.go:23]: why to capitalize the first letter

File: pkg/peermgr/swarm.go (hunk 1 above)
Comment (Karenlrx) [reply pkg/peermgr/swarm.go]: remove it

File: pkg/order/mempool/tx_cache.go (hunk 4 above)
Comment (Karenlrx) [reply pkg/order/mempool/tx_cache.go]: I'm a fool.    orz....

File: pkg/peermgr/swarm.go (hunk 2 above)
Comment (Karenlrx) [reply pkg/peermgr/swarm.go:222]: remove it

**Review by jzhe886 — APPROVED** [review]

File: pkg/order/etcdraft/node.go [hunk 5]
```diff
@@ -179,7 +182,8 @@ func NewNode(opts ...agency.ConfigOption) (agency.Order, error) {
 		storage:          dbStorage,
 		raftStorage:      raftStorage,
 		readyPool:        readyPool,
-		ctx:              context.Background(),
+		ctx:              ctx,
+		cancel: 		  cancel,
```
[the remark this answers is not in the document]
Comment (Karenlrx) [reply pkg/order/etcdraft/node.go]: It's a display problem on github

File: pkg/order/mempool/tx_cache.go [hunk 6]
```diff
@@ -111,3 +111,8 @@ func (tc *TxCache) stopTxSetTimer() {
 	close(tc.stopTimerC)
 	tc.stopTimerC = make(chan bool)
 }
+
+func (tc *TxCache) StopTxListen() {
+	defer close(tc.close)
```
Comment (sandyzhou) [inline pkg/order/mempool/tx_cache.go:116]: 是不是不需要往close channel发送true？直接close就可以了

Comment (Karenlrx) [reply pkg/order/mempool/tx_cache.go:116]: 实验了一下，close(ch)会从关闭的ch读取默认值（这里是false），已修改。

**Review by sandyzhou — APPROVED** [review]

**Pull request merged**